### ISSUE TITLE
Roll back a specific environment only 

### ISSUE THREAD

**Issue by maidul98** [Infisical/infisical#323]
Currently, we can only roll back secrets from the whole project and this includes changes from all environments. This means if I only want to roll back dev environment, it would also roll back all other environments including prod. Being able to roll back only a specific environment within a project would make the rollback feature more robust and useful.

**Comment by sheensantoscapadngan** [comment]
I can take this

**Comment by dangtony98** [comment]
This should be do-able.

It'd mean creating a new endpoint for that but the existing codebase should allow for it; you can check out the `/rollback` endpoints for example.

Another thing is rolling back a secret to a specific version.

**Comment by sheensantoscapadngan** [comment]
I came up with the ERD diagram below based on what I can infer from what we currently have. I marked in red the properties that we should consider adding for this development.

![Untitled Diagram drawio](https://github.com/Infisical/infisical/assets/65645666/8f0e7b94-cb19-4e9b-b099-30f757eb4fc4)

Some things to note based on the diagram:
1. We are already keeping track of what environment a secret or a secret version belongs to.
2. Adding an environment property to the Secret Snapshot model (marked in red) will allow us to support per environment grouping of secret versions which is what we need to support environment-based rollback (instead of workspace-wide)

I looked into our existing APIs and identified the following action points to do this:
1. Update the `GET api/v1/workspace/:workspaceId/secret-snapshots` endpoint so that it is required to provide an environment parameter in order to fetch the snapshots that are specific for a given workspace and environment
2. Update the `GET api/v1/workspace/:workspaceId/secret-snapshots/count` to require an environment parameter so that it only counts the snapshots of a given workspace and environment
3 Update the `POST api/v1/workspace/:workspaceId/secret-snapshots/rollback` so that it is required to provide an environment parameter in order to scope updates to a specific environment
4. Update helper function for creating SecretSnapshots so that it is scoped to making snapshots on a per environment basis. 

The endpoints for creating, deleting, batching, and updating secrets are already made such that requests are done on a per environment basis. No further changes have to be made other than the modification of the function needed to take a secret snapshot.

When I say update, it could mean just adding a new version for that API endpoint 👍
